src/bindable-property.js

```diff
@@ -35,7 +35,13 @@ export class BindableProperty {
   * Creates an instance of BindableProperty.
   * @param nameOrConfig The name of the property or a cofiguration object.
   */
-  constructor(nameOrConfig: string | Object) {
+  constructor(nameOrConfig: string | {
+    defaultBindingMode?: number,
+    reflect?: boolean | {(el: Element, name: string, newVal, oldVal): any},
```

Comment:
This is bike-shedding but I'm not a fan of the name. 
`reflect` is a little too generic for my taste, makes me think of `Reflection` first.
Polymer uses, `reflectToAttribute`, maybe we should play along? If someone sees this pop up in IntelliSense, its meaning is somewhat clear. `reflect` not as much.